src/fcntl.rs

```diff
@@ -153,21 +161,21 @@ libc_bitflags!(
 
 pub fn open<P: ?Sized + NixPath>(path: &P, oflag: OFlag, mode: Mode) -> Result<RawFd> {
     let fd = path.with_nix_path(|cstr| {
-        let modebits = c_uint::from(mode.bits());
-        unsafe { libc::open(cstr.as_ptr(), oflag.bits(), modebits) }
+        unsafe { libc::open(cstr.as_ptr(), oflag.bits(), mode.bits() as c_uint) }
```

Comment:
Why did you make this change?  Using `from` is usually preferable to `as`.

Reply:
It's because mode_t is a signed int on redox, so in terms of integer, because it can be lossy, the `From<c_uint>` trait is not implemented for `c_int` but `as c_uint` works fine.